radixdlt-core/radixdlt/src/test/java/com/radixdlt/network/p2p/addressbook/PeerPostponingManagerTest.java

```diff
@@ -0,0 +1,155 @@
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.network.p2p.addressbook;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.network.p2p.RadixNodeUri;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.Test;
+
+public class PeerPostponingManagerTest {
+  private final RadixNodeUri peer1 = createNodeUri("1.1.1.1");
+
+  @Test
+  public void peerCanRecordFailure() {
+    var manager =
+        new PeerPostponingManager(createClock(), Duration.ofMinutes(1), Duration.ofMinutes(3));
+
+    // First time it should survive, record is missing
+    assertTrue(manager.recordFailure(peer1));
+    // Second time it should survive, computed duration now is 2 min
+    assertTrue(manager.recordFailure(peer1));
+    // Should not survive, computed duration (4 min) is above limit (3 min)
+    assertFalse(manager.recordFailure(peer1));
+  }
+
+  @Test
+  public void successMakesPeerAccessible() {
+    var manager =
+        new PeerPostponingManager(createClock(), Duration.ofMinutes(1), Duration.ofMinutes(3));
+
+    assertTrue(manager.recordFailure(peer1));
+    assertTrue(manager.shouldIgnore(peer1));
+    manager.recordSuccess(peer1);
+    assertFalse(manager.shouldIgnore(peer1));
+  }
+
+  @Test
+  public void aliasesAreRemovedOnSuccessIfTheyAreExpired() {
+    var publicKey = ECKeyPair.generateNew().getPublicKey();
+    var peer1 = createNodeUri(publicKey, "1.1.1.1");
+    var peer2 = createNodeUri(publicKey, "1.1.1.2");
+
+    var manager =
+        new PeerPostponingManager(createClock(), Duration.ofMinutes(1), Duration.ofMinutes(3));
+
+    manager.recordFailure(peer1);
+    manager.recordFailure(peer2);
+
+    // Both addresses are postponed now
+    assertTrue(manager.shouldIgnore(peer1));
+    assertTrue(manager.shouldIgnore(peer2));
+
+    manager.recordSuccess(peer1);
+
+    // Both addresses are available now
+    assertFalse(manager.shouldIgnore(peer1));
```

Comment:
I'd rather the clock is explicitly set to be passed the 1 minute mark, rather than relying on the number of internal calls to `clock.instant()` to ensure the clock is sufficiently advanced to allow this.

Reply:
Reworked to use special kind of Clock.

Reply:
Really nice! 👍